flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecMultiJoin.java

```diff
@@ -0,0 +1,444 @@
+
+        return JoinUtil.generateConditionFunction(
+                config, classLoader, joinCondition, leftType, rightType);
+    }
+
+    private RowType leftTypeForIndex(List<RowType> inputRowTypes, int inputIndex) {
+        if (inputIndex <= 0) {
+            throw new IllegalArgumentException(
+                    "Input index must be greater than 0 for accumulated left type calculation");
+        }
+
+        RowType leftType = inputRowTypes.get(0);
+        for (int i = 1; i < inputIndex; i++) {
+            leftType = RowType.of(leftType, leftType);
+        }
+
+        return leftType;
+    }
+
+    /** Serializable factory for creating {@link StreamingMultiJoinOperator}. */
+    private static class StreamingMultiJoinOperatorFactoryImpl
+            extends AbstractStreamOperatorFactory<RowData> implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private final List<InternalTypeInfo<RowData>> inputTypeInfos;
+        private final List<JoinInputSideSpec> inputSideSpecs;
+        private final List<JoinType> joinTypes;
+        @Nullable private final MultiJoinCondition multiJoinCondition;
+        private final long[] stateRetentionTime;
+        private final JoinCondition[] joinConditions;
+        private final JoinKeyExtractor keyExtractor;
+        private final Map<
+                        Integer,
+                        Map<
+                                AttributeBasedJoinKeyExtractor.AttributeRef,
+                                AttributeBasedJoinKeyExtractor.AttributeRef>>
+                joinAttributeMap;
+
+        public StreamingMultiJoinOperatorFactoryImpl(
+                List<InternalTypeInfo<RowData>> inputTypeInfos,
+                List<JoinInputSideSpec> inputSideSpecs,
+                List<JoinType> joinTypes,
+                @Nullable MultiJoinCondition multiJoinCondition,
+                long[] stateRetentionTime,
+                JoinCondition[] joinConditions,
+                JoinKeyExtractor keyExtractor,
+                Map<
+                                Integer,
+                                Map<
+                                        AttributeBasedJoinKeyExtractor.AttributeRef,
+                                        AttributeBasedJoinKeyExtractor.AttributeRef>>
+                        joinAttributeMap) {
+            this.inputTypeInfos = inputTypeInfos;
+            this.inputSideSpecs = inputSideSpecs;
+            this.joinTypes = joinTypes;
+            this.multiJoinCondition = multiJoinCondition;
+            this.stateRetentionTime = stateRetentionTime;
+            this.joinConditions = joinConditions;
+            this.keyExtractor = keyExtractor;
+            this.joinAttributeMap = joinAttributeMap;
+        }
+
+        @Override
+        public <T extends StreamOperator<RowData>> T createStreamOperator(
+                StreamOperatorParameters<RowData> parameters) {
+            var inputRowTypes =
```

Comment:
```suggestion
            var inputRowTypes =
```
side comment: you are my first review that uses the `var` in Flink code base :D

Reply:
They're great :D I recommend it